### PR TITLE
Chore add testing

### PR DESCRIPTION
**Pull request by tomchentw**
Add test case to indicates that `TestUtils.Simulate` won't work for global elements. I also updated README.md to document this down.

I included some personal preferences for open source repos including `babel`, `eslint`, `travis` and `codeclimate`. Please let me know if you like some changes. I really appreciate your work on this library. :)

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "stage": 0
```
Comment (oliviertassinari) [inline .babelrc]: Do we need experimental feature of babel here?

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# Matches multiple files with brace expansion notation
+# Set default charset
+[*.{js,py}]
+charset = utf-8
+
+# 4 space indentation
+[*.py]
+indent_style = space
+indent_size = 4
```
Comment (oliviertassinari) [inline .editorconfig]: 2?

Comment (oliviertassinari) [inline .editorconfig]: Oh, it's for python.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+# Dependency directory
+# Commenting this out is preferred by some people, see
+# https://npmjs.org/doc/faq.html#Should-I-check-my-node_modules-folder-into-git
+node_modules
+
+# Users Environment Variables
+.lock-wscript
+
+
+### Project ###
```
Comment (oliviertassinari) [inline .gitignore]: I think that we don't need this section

**Comment by oliviertassinari** [comment]
Who, thanks for this! I'm closing and openning the PR to trigger the travis tests.

**Comment by oliviertassinari** [comment]
@tomchentw I'm gonna merge this, we can clean it later

File: .babelrc (hunk 1 above)
Comment (tomchentw) [inline .babelrc]: Just my personal pref.

File: .gitignore (hunk 3 above)
Comment (tomchentw) [inline .gitignore]: Yeah feel free to remove this section.

**Comment by tomchentw** [comment]
Thanks for the merge. Sorry I didn't reply immediately.

Couple of things to do on badges:
1. Create a chat room on https://gitter.im
2. Setup coverage token for travis:

- First, log in into codeclimate: https://codeclimate.com/github/oliviertassinari/react-event-listener
- Then go to _Coverage_ section, __JavaScript__ tab
- You'll find a `CODECLIMATE_REPO_TOKEN` stuff and we need to add it to `.travis.yml`
- However, we don't directly add it. Instead, we have to encrypt it. See http://docs.travis-ci.com/user/encryption-keys/
- `travis encrypt CODECLIMATE_REPO_TOKEN=1234567890qazxwedcvggfrtgbnhyujm --add` (This token is fake)
- Commit the changes made to `.travis.yml` and push up. Now travis CI will run the test cases again and upload coverage result to codeclimate.
- And we should see the badges updated there.

**Pull request merged**